src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -196,39 +197,49 @@ public void doXContent(XContentBuilder builder, Params params) throws IOExceptio
     }
 
     @Override
-    protected Query doToQuery(QueryShardContext context) throws IOException {
-
+    protected Query doToQuery(QueryShardContext context) {
         MappedFieldType mappedFieldType = context.fieldMapper(this.fieldName);
 
         if (!(mappedFieldType instanceof KNNVectorFieldMapper.KNNVectorFieldType)) {
-            throw new IllegalArgumentException("Field '" + this.fieldName + "' is not knn_vector type.");
+            throw new IllegalArgumentException(String.format("Field '%s' is not knn_vector type.", this.fieldName));
         }
 
-        int dimension = ((KNNVectorFieldMapper.KNNVectorFieldType) mappedFieldType).getDimension();
-
-        // If the dimension is not set, then the only valid route forward is if the field uses a model
-        if (dimension == -1) {
-            String modelId = ((KNNVectorFieldMapper.KNNVectorFieldType) mappedFieldType).getModelId();
-
-            if (modelId == null) {
-                throw new IllegalArgumentException("Field '" + this.fieldName + "' does not have dimension set.");
-            }
-
-            ModelMetadata modelMetadata = modelDao.getMetadata(modelId);
+        int fieldDimension = ((KNNVectorFieldMapper.KNNVectorFieldType) mappedFieldType).getDimension();
+        KNNMethodContext knnMethodContext = ((KNNVectorFieldMapper.KNNVectorFieldType) mappedFieldType).getKnnMethodContext();
```

Comment:
nit: we can create variable of correct type 
```suggestion
        KNNVectorFieldMapper.KNNVectorFieldType knnVectorMappedFieldType (KNNVectorFieldMapper.KNNVectorFieldType) mappedFieldType;
        int fieldDimension = knnVectorMappedFieldType.getDimension();
        KNNMethodContext knnMethodContext = knnVectorMappedFieldType.getKnnMethodContext();
```